src/Core/Managed/Shared/ActivityExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿#if !NET40
+namespace Microsoft.ApplicationInsights
+{
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Runtime.CompilerServices;
+
+    internal static class ActivityExtensions
+    {
+        private const string OperationNameTag = "OperationName";
+        private static bool isInitialized = false;
+        private static bool isEnabled = false;
+
+        /// <summary>
+        /// Checks that Activity API could be called (DiagnosticSource DLL is loaded).
+        /// </summary>
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)] 
+        public static bool IsActivityEnabled()
+        {
+            if (isInitialized)
+            {
+                return isEnabled;
+            }
+
+            try
+            {
+                var activity = new Activity("IsEnabled");
+                isEnabled = true;
+                isInitialized = true;
+                return true;
+            }
+            catch (System.IO.FileNotFoundException)
```

Comment:
In fact, BindingFlags that are needed for Activator.CreateInstance are not available in netstandard 1.3. So, I'll call Assembly.Load